app/views/finance/invoices/show.html.haml

```diff
@@ -35,6 +35,7 @@
   %b= heading_helper(Invoice, :note) + ':'
   =h @invoice.note
 
-= link_to t('ui.edit'), edit_finance_invoice_path(@invoice)
-|
+- if current_user.role_finance? or (current_user.role_invoices? and not @invoice.paid_on)
```

Comment:
where should I put it? (and maybe a name for it ;-)) IMHO it's not a member function of User or Invoice...